tasks/firewall.yml

```diff
@@ -40,30 +20,12 @@
             or
             'fence-agents-all' in ha_cluster_extra_packages
           )
-      block:
-        - name: Enable fence-virt port in firewalld
-          command: firewall-cmd --add-port 1229/tcp
-          when: ansible_facts.services['firewalld.service'].state == 'running'
-          register: __ha_cluster_firewall_cmd
-          failed_when:
-            - __ha_cluster_firewall_cmd.rc != 0
-            - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
-        - name: Enable fence-virt port in firewalld permanent config
-          command: firewall-cmd --permanent --add-port 1229/tcp
-          when: ansible_facts.services['firewalld.service'].state == 'running'
-          register: __ha_cluster_firewall_cmd
-          failed_when:
-            - __ha_cluster_firewall_cmd.rc != 0
-            - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
-        - name: Enable fence-virt port in firewalld offline config
-          command: firewall-offline-cmd --add-port 1229/tcp
-          when: ansible_facts.services['firewalld.service'].state != 'running'
-          register: __ha_cluster_firewall_cmd
-          failed_when:
-            - __ha_cluster_firewall_cmd.rc != 0
-            - __ha_cluster_firewall_cmd.rc != 11  # already enabled
 
-  when:
-    - '"firewalld.service" in ansible_facts.services'
+    - name: Ensure the service and the ports status with the firewall role
+      include_role:
+        name: fedora.linux_system_roles.firewall
+      vars:
+        firewall: "{{ _ha_cluster_firewall }}"
+  vars:
+    _state_value: "{{ 'enabled' if ha_cluster_manage_firewall |
+                   bool | d(false) else 'disabled' }}"
```

Comment:
Why do you need `d(false)` here if the ha_cluster_manage_firewall is set to `false` in defaults/main.yml?